### PR TITLE
Adding contextual View for collection overviews

### PR DESCRIPTION
**Pull request by knjackson30974**
Adding contextual View for collection overviews

### CODE REVIEW

File: views/islandora_usage_stats_collections.views_default.inc [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<?php
+$view = new view();
```
Comment (adam-vessey) [inline views/islandora_usage_stats_collections.views_default.inc]: This code (all the default view code, that is) should live in an implementation of [`hook_views_default_views()`](https://api.drupal.org/api/views/views.api.php/function/hook_views_default_views/7).

File: islandora_usage_stats.module [hunk 2]
```diff
@@ -564,3 +561,14 @@ function islandora_usage_stats_menu_alter(&$items) {
   $items['islandora/object/%islandora_object/datastream/%islandora_datastream/download']['module']
     = "islandora_usage_stats";
 }
+/**
+ * Implements hook_islandora_overview_object_alter().
+ *
+ */
+function islandora_usage_stats_islandora_collectioncmodel_islandora_overview_object_alter(AbstractObject &$object, &$output) {
+  // The key here doesn't really matter... Should just try to make sure it's unique (module prefixing helps this).
```
Comment (adam-vessey) [inline islandora_usage_stats.module]: My comment had been intended for you... Not to make it into the code. :P

File: views/islandora_usage_stats_collections.views_default.inc [hunk 3]
```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Implements hook_views_default_views().
+ */
+function islandora_usage_stats_collection_views_default_views() {
```
Comment (adam-vessey) [inline views/islandora_usage_stats_collections.views_default.inc]: The name of the function should just be `islandora_usage_stats_views_default_views`.

File: views/islandora_usage_stats_collections.views_default.inc [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    t('Object Collection Access'),
+    t('Label'),
+    t('PID'),
+    t('Log ID'),
+    t('.'),
+    t(','),
+    t('All'),
+    t('Block'),
+    t('Usage Stats for Collections'),
+  );
```
Comment (adam-vessey) [inline views/islandora_usage_stats_collections.views_default.inc]: This function should be returning an associative array of these built up "view" objects. Maybe look at [islandora_batch](https://github.com/Islandora/islandora_batch) for [an example](https://github.com/Islandora/islandora_batch/blob/7.x/views/islandora_batch.views_default.inc#L11-L462)?

File: views/islandora_usage_stats_collections.views_default.inc (hunk 3 above)
Comment (adam-vessey) [inline views/islandora_usage_stats_collections.views_default.inc]: Actually... The code in this function should just be added to https://github.com/discoverygarden/islandora_usage_stats/blob/7.x/views/islandora_usage_stats.views_default.inc, and another entry created in the array.

Comment (knjackson30974) [inline views/islandora_usage_stats_collections.views_default.inc]: I added the view to the usage_stats as another array entry based on the islandora_batch.

File: islandora_usage_stats.module (hunk 2 above)
Comment (knjackson30974) [inline islandora_usage_stats.module]: Haha whoops... thought I took that out.

File: views/islandora_usage_stats_collections.views_default.inc [hunk 5]
```diff
@@ -0,0 +1,145 @@
+<?php
```
Comment (adam-vessey) [inline views/islandora_usage_stats_collections.views_default.inc]: This file should probably be deleted, now.

File: islandora_usage_stats.module [hunk 6]
```diff
@@ -564,3 +561,12 @@ function islandora_usage_stats_menu_alter(&$items) {
   $items['islandora/object/%islandora_object/datastream/%islandora_datastream/download']['module']
     = "islandora_usage_stats";
 }
+/**
+ * Implements hook_islandora_overview_object_alter().
+ */
+function islandora_usage_stats_collections_islandora_overview_object_alter(AbstractObject &$object, &$output) {
```
Comment (adam-vessey) [inline islandora_usage_stats.module]: The `collections` bit in this function name would(/should) cause it to never be matched... `islandora:collectionCModel` is the content model we want to target, yeah? To make a function name, we transform "invalid" characters (`:`, `-`, _etc_) to underscores (and PHP function names are not case sensitive)... So to have this alter happen only on objects with the `islandora:collectionCModel` model, the function should likely be named `islandora_usage_stats_islandora_collectioncmodel_islandora_overview_object_alter`.

File: islandora_usage_stats.module [hunk 7]
```diff
@@ -107,10 +106,8 @@ function islandora_usage_stats_block_configure($delta) {
           '#default_value' => variable_get('islandora_usage_stats_top_search_block_display_rows', 5),
         );
       }
-      break;
-
+    break;
```
Comment (adam-vessey) [inline islandora_usage_stats.module]: Looks like the indentation here was affected for the worse:

```
sites/all/modules/islandora_usage_stats/islandora_usage_stats.module:
 +109: [minor] Case breaking statement indented incorrectly; expected 6 spaces, found 4
```

File: views/islandora_usage_stats.views_default.inc [hunk 8]
```diff
@@ -336,5 +338,150 @@ function islandora_usage_stats_views_default_views() {
     t('Label'),
   );
 
-  return array('islandora_usage_stats_report' => $view);
+  $views['islandora_usage_stats_report'] = $view;
+
```
Comment (adam-vessey) [inline views/islandora_usage_stats.views_default.inc]: Not overly concerned, but yeah:
```
sites/all/modules/islandora_usage_stats/views/islandora_usage_stats.views_default.inc:
 +342: [minor] Functions must not contain multiple empty lines in a row; found 2 empty lines
```

File: views/islandora_usage_stats.views_default.inc [hunk 9]
```diff
@@ -336,5 +338,149 @@ function islandora_usage_stats_views_default_views() {
...
+  $view = new view();
```
Comment (adam-vessey) [inline views/islandora_usage_stats.views_default.inc]: With a few more options enabled (sortable columns, column titles, pager, _etc_):
```php
$view = new view();
$view->name = 'usage_collection';
$view->description = 'Usage Stats for Collections';
$view->tag = 'default';
$view->base_table = 'islandora_usage_stats_object_access_log';
$view->human_name = 'Islandora Usage Stats for Collections';
$view->core = 7;
$view->api_version = '3.0';
$view->disabled = FALSE; /* Edit this to true to make a default view disabled initially */

/* Display: Master */
$handler = $view->new_display('default', 'Master', 'default');
$handler->display->display_options['title'] = 'Usage Stats';
$handler->display->display_options['use_more_always'] = FALSE;
$handler->display->display_options['group_by'] = TRUE;
$handler->display->display_options['access']['type'] = 'perm';
$handler->display->display_options['cache']['type'] = 'none';
$handler->display->display_options['query']['type'] = 'views_query';
$handler->display->display_options['exposed_form']['type'] = 'basic';
$handler->display->display_options['pager']['type'] = 'full';
$handler->display->display_options['pager']['options']['items_per_page'] = '10';
$handler->display->display_options['pager']['options']['offset'] = '0';
$handler->display->display_options['pager']['options']['id'] = '0';
$handler->display->display_options['pager']['options']['quantity'] = '9';
$handler->display->display_options['style_plugin'] = 'table';
$handler->display->display_options['style_options']['columns'] = array(
  'label' => 'label',
  'pid' => 'pid',
  'id' => 'id',
);
$handler->display->display_options['style_options']['default'] = 'id';
$handler->display->display_options['style_options']['info'] = array(
  'label' => array(
    'sortable' => 1,
    'default_sort_order' => 'asc',
    'align' => '',
    'separator' => '',
    'empty_column' => 0,
  ),
  'pid' => array(
    'sortable' => 1,
    'default_sort_order' => 'asc',
    'align' => '',
    'separator' => '',
    'empty_column' => 0,
  ),
  'id' => array(
    'sortable' => 1,
    'default_sort_order' => 'desc',
    'align' => '',
    'separator' => '',
    'empty_column' => 0,
  ),
);
/* Header: Global: Result summary */
$handler->display->display_options['header']['result']['id'] = 'result';
$handler->display->display_options['header']['result']['table'] = 'views';
$handler->display->display_options['header']['result']['field'] = 'result';
$handler->display->display_options['header']['result']['empty'] = TRUE;
/* No results behavior: Global: Text area */
$handler->display->display_options['empty']['area']['id'] = 'area';
$handler->display->display_options['empty']['area']['table'] = 'views';
$handler->display->display_options['empty']['area']['field'] = 'area';
$handler->display->display_options['empty']['area']['empty'] = TRUE;
$handler->display->display_options['empty']['area']['content'] = 'No view counts found.';
$handler->display->display_options['empty']['area']['format'] = 'filtered_html';
/* Relationship: Islandora Usage Stats: Collection */
$handler->display->display_options['relationships']['id']['id'] = 'id';
$handler->display->display_options['relationships']['id']['table'] = 'islandora_usage_stats_object_access_log';
$handler->display->display_options['relationships']['id']['field'] = 'id';
/* Relationship: Islandora Usage Stats: Object */
$handler->display->display_options['relationships']['pid_id']['id'] = 'pid_id';
$handler->display->display_options['relationships']['pid_id']['table'] = 'islandora_usage_stats_object_access_log';
$handler->display->display_options['relationships']['pid_id']['field'] = 'pid_id';
/* Relationship: Islandora Usage Stats: Collection Object */
$handler->display->display_options['relationships']['collection']['id'] = 'collection';
$handler->display->display_options['relationships']['collection']['table'] = 'islandora_usage_stats_collection_access_log';
$handler->display->display_options['relationships']['collection']['field'] = 'collection';
$handler->display->display_options['relationships']['collection']['relationship'] = 'id';
/* Relationship: Islandora Usage Stats: Object Collection Access */
$handler->display->display_options['relationships']['object_access_id']['id'] = 'object_access_id';
$handler->display->display_options['relationships']['object_access_id']['table'] = 'islandora_usage_stats_collection_access_log';
$handler->display->display_options['relationships']['object_access_id']['field'] = 'object_access_id';
$handler->display->display_options['relationships']['object_access_id']['relationship'] = 'id';
/* Field: Islandora Usage Stats: Label */
$handler->display->display_options['fields']['label']['id'] = 'label';
$handler->display->display_options['fields']['label']['table'] = 'islandora_usage_stats_objects';
$handler->display->display_options['fields']['label']['field'] = 'label';
$handler->display->display_options['fields']['label']['relationship'] = 'pid_id';
/* Field: Islandora Usage Stats: PID */
$handler->display->display_options['fields']['pid']['id'] = 'pid';
$handler->display->display_options['fields']['pid']['table'] = 'islandora_usage_stats_objects';
$handler->display->display_options['fields']['pid']['field'] = 'pid';
$handler->display->display_options['fields']['pid']['relationship'] = 'pid_id';
$handler->display->display_options['fields']['pid']['alter']['make_link'] = TRUE;
$handler->display->display_options['fields']['pid']['alter']['path'] = 'islandora/object/[pid]';
$handler->display->display_options['fields']['pid']['element_label_colon'] = FALSE;
/* Field: COUNT(DISTINCT Islandora Usage Stats: Log ID) */
$handler->display->display_options['fields']['id']['id'] = 'id';
$handler->display->display_options['fields']['id']['table'] = 'islandora_usage_stats_object_access_log';
$handler->display->display_options['fields']['id']['field'] = 'id';
$handler->display->display_options['fields']['id']['group_type'] = 'count_distinct';
$handler->display->display_options['fields']['id']['label'] = 'Page Views';
/* Contextual filter: Islandora Usage Stats: PID */
$handler->display->display_options['arguments']['pid']['id'] = 'pid';
$handler->display->display_options['arguments']['pid']['table'] = 'islandora_usage_stats_objects';
$handler->display->display_options['arguments']['pid']['field'] = 'pid';
$handler->display->display_options['arguments']['pid']['relationship'] = 'collection';
$handler->display->display_options['arguments']['pid']['default_action'] = 'default';
$handler->display->display_options['arguments']['pid']['default_argument_type'] = 'raw';
$handler->display->display_options['arguments']['pid']['default_argument_options']['index'] = '2';
$handler->display->display_options['arguments']['pid']['summary']['number_of_records'] = '0';
$handler->display->display_options['arguments']['pid']['summary']['format'] = 'default_summary';
$handler->display->display_options['arguments']['pid']['summary_options']['items_per_page'] = '25';
$handler->display->display_options['arguments']['pid']['limit'] = '0';

/* Display: Block */
$handler = $view->new_display('block', 'Block', 'usage_stats');
$handler->display->display_options['block_description'] = 'Usage Stats for Collections';
$translatables['usage_collection'] = array(
  t('Master'),
  t('Usage Stats'),
  t('more'),
  t('Apply'),
  t('Reset'),
  t('Sort by'),
  t('Asc'),
  t('Desc'),
  t('Items per page'),
  t('- All -'),
  t('Offset'),
  t('« first'),
  t('‹ previous'),
  t('next ›'),
  t('last »'),
  t('Displaying @start - @end of @total'),
  t('No view counts found.'),
  t('Collection'),
  t('Object'),
  t('Collection Object'),
  t('Object Collection Access'),
  t('Label'),
  t('PID'),
  t('Page Views'),
  t('.'),
  t(','),
  t('All'),
  t('Block'),
  t('Usage Stats for Collections'),
);
```

**Pull request merged**